types/proposal.go

```diff
@@ -169,6 +179,19 @@ func (p *Proposal) ToProto() *tmproto.Proposal {
 	pb.PolRound = p.POLRound
 	pb.Timestamp = p.Timestamp
 	pb.Signature = p.Signature
+	var txKeys []*tmproto.TxKey
```

Comment:
nit: could make use of knowing the txKeys size here `txKeys := make([]*tmproto.TxKey, 0, len(txKeys))`